save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -25,21 +27,38 @@ data class Warning(
  * @param lineGroupIdx index of capture group for line number
  * @param messageGroupIdx index of capture group for waring text
  * @return a [Warning] or null if [this] string doesn't match [warningRegex]
+ * @throws ResourceFormatException when parsing a file
  */
 internal fun String.extractWarning(warningRegex: Regex,
                                    columnGroupIdx: Int?,
                                    lineGroupIdx: Int?,
                                    messageGroupIdx: Int,
 ): Warning? {
     val groups = warningRegex.find(this)?.groups ?: return null
+
     val line = lineGroupIdx?.let {
-        groups[lineGroupIdx]?.value?.toInt()
+        try {
+            groups[lineGroupIdx]!!.value.toInt()
+        } catch (e: Exception) {
+            throw ResourceFormatException("Invalid line format in test file: $this")
```

Comment:
```suggestion
            throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.message}")
```
I think this would be even more informative, please change other exception texts similarly